runners/samza/src/main/java/org/apache/beam/runners/samza/translation/TranslationContext.java

```diff
@@ -145,9 +152,25 @@ protected <KeyT, OutT> void registerInputMessageStreams(
   }
 
   public <OutT> void registerMessageStream(PValue pvalue, MessageStream<OpMessage<OutT>> stream) {
+    registerMessageStream(pvalue, stream, true);
+  }
+
+  public <OutT> void registerMessageStream(
+      PValue pvalue, MessageStream<OpMessage<OutT>> stream, boolean enableTransformMetric) {
     if (messsageStreams.containsKey(pvalue)) {
       throw new IllegalArgumentException("Stream already registered for pvalue: " + pvalue);
     }
+    // add a step to attach OutputMetricOp if registered for Op Stream
+    final Config overrideConfig = new MapConfig(getPipelineOptions().getConfigOverride());
```

Comment:
Don't use config override. Use PipelineOptions. We have an option for enable/disable metrics, pls add one more for TransformMetrics.